src/python/hooks/sealights_test.go

```diff
@@ -0,0 +1,333 @@
+			Expect(ModifiedCommand).To(Equal("web: sl-python run --tokenfile some-token-file --buildsessionidfile some-bsid-file --proxy some-proxy --labid some-labid --  python flask.py"))
+			Expect(err).To(BeNil())
+		})
+		It("Returns an error when provided the wrong format", func() {
+			slConfig := hooks.NewSealightsConfig()
+			startCommand := "python flask.py"
+			_, err := sealights.GenerateStartUpCommand(startCommand, slConfig.GetStartFlags())
+			Expect(err).To(MatchError("improper format found in Procfile"))
+		})
+
+	})
+
+	Context("RewriteProcFile", func() {
+		var (
+			err         error
+			tempProcDir string
+			slConfig    *hooks.SealightsConfig
+		)
+		BeforeEach(func() {
+			tempProcDir, err = os.MkdirTemp("", "Procfiles")
+			Expect(err).To(BeNil())
+			slConfig = hooks.NewSealightsConfig()
+		})
+
+		AfterEach(func() {
+			Expect(os.RemoveAll(tempProcDir)).To(Succeed())
+		})
+
+		It("rewrites the procfile with sl-python", func() {
+			err = createNewFile(tempProcDir, "Procfile", "web: python app.py", 0666)
+			Expect(err).To(BeNil())
+			err = sealights.RewriteProcFile(filepath.Join(tempProcDir, "Procfile"), slConfig.GetStartFlags())
+			Expect(err).To(BeNil())
+			startCommand, err := os.ReadFile(filepath.Join(tempProcDir, "Procfile"))
+			Expect(err).To(BeNil())
+			Expect(string(startCommand)).To(Equal("web: sl-python run --  python app.py"))
+		})
+
+		It("Errors when Procfile doesn't exist", func() {
+			err := sealights.RewriteProcFile("/doesnt/exist", slConfig.GetStartFlags())
+			Expect(err).To(MatchError("Error reading file /doesnt/exist: open /doesnt/exist: no such file or directory"))
+		})
+
+		It("Errors with Procfile with wrong format", func() {
+			err = createNewFile(tempProcDir, "WrongFormatProcFile", "python app.py", 0666)
+			Expect(err).To(BeNil())
+
+			err = sealights.RewriteProcFile(filepath.Join(tempProcDir, "WrongFormatProcFile"), slConfig.GetStartFlags())
+			Expect(err).To(MatchError("improper format found in Procfile"))
+		})
+	})
+	Context("RewriteRequirementsFile when requirements.txt is not packaged", func() {
+		It("creates requirements.txt", func() {
+			Expect(libbuildpack.FileExists(filepath.Join(buildDir, "requirements.txt"))).ToNot(BeTrue())
+			err := sealights.RewriteRequirementsFile(stager, "")
+			Expect(err).To(BeNil())
+			Expect(libbuildpack.FileExists(filepath.Join(buildDir, "requirements.txt"))).To(BeTrue())
+			packagesList, err := os.ReadFile(filepath.Join(buildDir, "requirements.txt"))
+			Expect(err).To(BeNil())
+			Expect(string(packagesList)).To(Equal("sealights-python-agent"))
+		})
+	})
+
+	Context("RewriteRequirementsFile when requirements.txt is packaged", func() {
+		BeforeEach(func() {
+			Expect(libbuildpack.FileExists(filepath.Join(buildDir, "requirements.txt"))).ToNot(BeTrue())
+			procFile := filepath.Join(buildDir, "requirements.txt")
+			f, err := os.OpenFile(procFile, os.O_CREATE|os.O_WRONLY, 0644)
```

Comment:
I think it would be way simpler to replace all this file opening and writing logic with:

```
requirementsFilepath := filepath.Join(buildDir, "requirements.txt")
Expect(libbuildpack.FileExists(requirementsFilepath).ToNot(BeTrue())

Expect(os.WriteFile(requirementsFilepath, []byte("Flask"), 0644)).To(Succeed())
```